odoo/fields.py

```diff
@@ -2967,14 +2990,14 @@ def convert_to_display_name(self, value, record):
         return value.display_name if value else False
 
 
-class _Relational(Field):
+class _Relational(Field[M], typing.Generic[M]):
```

Comment:
Before removing, I'll have to check how it will bind `M` when typechecking if `Generic` is not there.

Reply:
After checking, you can remove `Generic` since we force users to overwrite the typing information on the field itself. You may leave it on the abstract/private field classes.